### DISCUSSION TITLE
Zustand doesn't trigger component update when state is class based

### DISCUSSION THREAD

**Discussion by pablojakub** [pmndrs/zustand#2539] [Q&A]
Hello,
I want to store objects in Zustand created from class: 

`const initialState: InitialState = {
    animals: Animal[],
};`

Store is feeded by pushing new classess to the collection of animals. Whenever I update any object it doesn't trigger components update. Neither using immer nor typical set state: 

`updateAnimal: (animalld) => set((state: AnimalSlice) => {
                const newAnimals = [...state.animalSlice.animals];
                const animalToUpdate = newAnimals .find((animal) => animal.id === animalId);
                const animalToUpdateIndex = newAnimals .findIndex((animal) => animal.id === animalId);
                if (animalToUpdate ) {
                    animalToUpdate .name = 'Cow';
                    newAnimals [animalToUpdateIndex ] = animalToUpdate ;
                }
                return {
                    animalSlice: { actions: { ...state.animalSlice.actions }, selectors: { ...state.animalSlice.selectors }, animals: newAnimals },
                };
            }),`

In devtools I can see that name changed to "Cow" but it doesn't trigger rerender of given component.

Can anyone has that kind of situation?

**Comment by dbritto-dev** [comment]
@pablojakub don't use classes don't do that to you
